sensor/kubernetes/listener/resources/convert.go

```diff
@@ -273,10 +273,16 @@ func (w *deploymentWrap) getPods(hierarchy references.ParentHierarchy, labelSele
 
 func (w *deploymentWrap) populateDataFromPods(pods ...*v1.Pod) {
 	w.pods = pods
-	w.populateImageIDs(pods...)
+	// Sensor must already know about the OpenShift internal registries to determine if an image is cluster-local,
+	// which is ok because Sensor listens for Secrets before it starts listening for Deployment-like resources.
+	w.populateImageMetadata(registry.Singleton(), pods...)
```

Comment:
This dependency should ideally be injected too, but if you decide it's too painful, that's fine (since you're not making it worse than it was before in this respect)

Reply:
done